bodo/pandas/_physical_conv.cpp

```diff
@@ -52,9 +56,90 @@ void PhysicalPlanBuilder::Visit(duckdb::LogicalProjection& op) {
+    switch (type) {
+        case duckdb::LogicalTypeId::INTEGER:
+            return value.GetValue<int>();
+        case duckdb::LogicalTypeId::FLOAT:
+            return value.GetValue<float>();
+        case duckdb::LogicalTypeId::DOUBLE:
+            return value.GetValue<double>();
+        default:
+            throw std::runtime_error("extractValue unhandled type.");
+    }
+}
+
+std::shared_ptr<PhysicalExpression> buildPhysicalExprTree(
+    duckdb::unique_ptr<duckdb::Expression>& expr) {
+    duckdb::ExpressionClass expr_class = expr->GetExpressionClass();
+    duckdb::ExpressionType expr_type = expr->GetExpressionType();
+
+    switch (expr_class) {
+        case duckdb::ExpressionClass::BOUND_COMPARISON: {
+            auto bce =
+                dynamic_cast_unique_ptr<duckdb::BoundComparisonExpression>(
+                    expr);
+            return std::static_pointer_cast<PhysicalExpression>(
+                std::make_shared<PhysicalComparisonExpression>(
+                    buildPhysicalExprTree(bce->left),
+                    buildPhysicalExprTree(bce->right), expr_type));
+        } break;  // suppress wrong fallthrough error
+        case duckdb::ExpressionClass::BOUND_COLUMN_REF: {
+            auto bce =
+                dynamic_cast_unique_ptr<duckdb::BoundColumnRefExpression>(expr);
+            auto binding = bce->binding;
+            return std::static_pointer_cast<PhysicalExpression>(
+                std::make_shared<PhysicalColumnRefExpression>(
+                    binding.table_index, binding.column_index));
+        } break;  // suppress wrong fallthrough error
+        case duckdb::ExpressionClass::BOUND_CONSTANT: {
+            auto bce =
+                dynamic_cast_unique_ptr<duckdb::BoundConstantExpression>(expr);
+            auto extracted_value = extractValue(bce->value);
+            return std::visit(
+                [](const auto& value) {
+                    return std::static_pointer_cast<PhysicalExpression>(
+                        std::make_shared<PhysicalConstantExpression<
+                            std::decay_t<decltype(value)>>>(value));
+                },
+                extracted_value);
+        } break;  // suppress wrong fallthrough error
+        default:
+            throw std::runtime_error(
+                "Unsupported duckdb expression type" +
+                std::to_string(static_cast<int>(expr_class)));
+    }
+    throw std::logic_error("Control should never reach here");
+}
+
 void PhysicalPlanBuilder::Visit(duckdb::LogicalFilter& op) {
-    throw std::runtime_error(
-        "Not supported on the physical side yet: LogicalFilter");
+    // Process the source of this projection.
+    this->Visit(*op.children[0]);
+
+    if (op.expressions.size() != 1) {
+        throw std::runtime_error(
+            "LogicalFilter only supports expressions of size 1");
+    }
+
+    auto physExprTree = buildPhysicalExprTree(op.expressions[0]);
```

Comment:
Minor: not the best use of `auto` since output type isn't clear here (as opposed to the next line with `make_shared`).

Reply:
Done.